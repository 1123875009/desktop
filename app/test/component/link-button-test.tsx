import * as React from 'react'
import { render, cleanup, fireEvent } from 'react-testing-library'

import { LinkButton } from '../../src/ui/lib/link-button'

describe('<LinkButton/>', () => {
  afterEach(cleanup)

  it(`renders an element without the link`, () => {
    const { container } = render(<LinkButton>some text goes here</LinkButton>)
    expect(container).toMatchSnapshot()
  })

  it(`renders an element with a link`, () => {
    const { container } = render(
      <LinkButton uri="https://www.google.com/">
        this link contains buttons
      </LinkButton>
    )
    expect(container).toMatchSnapshot()
  })

  it(`can click element to trigger callback`, () => {
    const clickEvent = jest.fn()
    const { getByText } = render(
      <LinkButton onClick={clickEvent}>click me</LinkButton>
    )

    fireEvent.click(getByText('click me'))
    expect(clickEvent).toHaveBeenCalledTimes(1)
  })
})
