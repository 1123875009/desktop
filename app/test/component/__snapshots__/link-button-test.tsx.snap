// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkButton/> renders an element with a link 1`] = `
<div>
  <a
    class="link-button-component"
    href="https://www.google.com/"
  >
    this link contains buttons
  </a>
</div>
`;

exports[`<LinkButton/> renders an element without the link 1`] = `
<div>
  <a
    class="link-button-component"
    href=""
  >
    some text goes here
  </a>
</div>
`;
